drivers/entropy/entropy_nrf5.c

```diff
@@ -269,53 +241,55 @@ static int entropy_nrf5_get_entropy(struct device *device, u8_t *buf, u16_t len)
 static int entropy_nrf5_get_entropy_isr(struct device *dev, u8_t *buf, u16_t len,
 					u32_t flags)
 {
-	struct entropy_nrf5_dev_data *dev_data = DEV_DATA(dev);
 	u16_t cnt = len;
 
-	if (!(flags & ENTROPY_BUSYWAIT)) {
-		return get((struct rand *)dev_data->isr, len, buf);
+	/* Check if this API is called on correct driver instance. */
+	__ASSERT_NO_MSG(&entropy_nrf5_data == DEV_DATA(dev));
```

Comment:
I thought you wanted this to be fast. And now with this assertion it will be even slower than in previous version.

Reply:
I'd rather get rid of the `DEV_DATA` macro completely (not use the `driver_data` field from the `device` structure at all) to indicate somehow that this driver handles one instance only.

Reply:
Normally asserts are not compiled in. I would like to keep the asserts in order to check if our assumptions are still correct.